src/system_info/system_info_utils.cc

```diff
@@ -48,7 +48,93 @@ char* GetDuidProperty() {
   }
   return s;
 }
-#endif
+
+#ifndef TIZEN_MOBILE
+GDBusConnection* GetDbusConnection() {
+  GError* error = NULL;
+  gchar* addr = g_dbus_address_get_for_bus_sync(G_BUS_TYPE_SYSTEM,
+                                                NULL, &error);
+  if (!addr) {
+    std::cout << "fail to get dbus addr: " << error->message << std::endl;
+    g_free(error);
+    return NULL;
+  }
+
+  GDBusConnection* bus_conn = g_dbus_connection_new_for_address_sync(addr,
+      (GDBusConnectionFlags)(G_DBUS_CONNECTION_FLAGS_AUTHENTICATION_CLIENT |
+      G_DBUS_CONNECTION_FLAGS_MESSAGE_BUS_CONNECTION), NULL, NULL, &error);
+  if (!bus_conn) {
+    std::cout << "fail to create dbus connection: "
+              << error->message << std::endl;
+    g_free(error);
+    return NULL;
+  }
+  return bus_conn;
+}
+
+char* OfonoGetModemPath(GDBusConnection* bus_conn) {
+  GError* error = NULL;
+  GVariant* modems_result = g_dbus_connection_call_sync(bus_conn,
+      kOfonoService, kOfonoManagerPath, kOfonoManagerIface,
+      "GetModems", NULL, NULL, G_DBUS_CALL_FLAGS_NONE, -1, NULL, &error);
+
+  if (!modems_result) {
+    std::cout << "ofono GetModems failed: " << error->message << std::endl;
+    g_error_free(error);
+    return NULL;
+  }
+
+  char* result_path = NULL;
+  GVariantIter* modems_iter;
+  g_variant_get(modems_result, "(a(oa{sv}))", &modems_iter);
+  char* path;
+  while (g_variant_iter_next(modems_iter, "(o@a{sv})", &path, NULL)) {
```

Comment:
"(o@a{sv})" indicates that a{sv} is a ````GVariant*```` type, g_variant_iter_next() needs a ````GVariant*```` type to put the value, here NULL replaced as it's no use.
"(oa{sv})" meas a{sv} is a ````Dictionaries```` type, g_variant_iter_next need() a ````GVariantIter*```` type to put the value, NULL is OK, too.
So, in this case both "(o@a{sv})" and "(oa{sv})" is workable.
 @zolkis you give me an inspiration -- I can get type value form modems_properties!
````
GVariant* modems_properties;
g_variant_iter_next(modems_iter, "(o@a{sv})", &path, &modems_properties)
g_variant_get(modems_properties, "a{sv}", &iter)
........
````